code/game/objects/structures/crates_lockers/wall_closets.dm

```diff
@@ -0,0 +1,285 @@
+/obj/structure/wall_closet
+	name = "wall closet"
+	desc = "It's a basic, wall mounted, storage unit."
+	anchored = TRUE
+	icon = 'icons/obj/storage/wall_closet.dmi'
+	icon_state = "generic"
+	var/theme_color = "#5f5f5f"
+	var/list/closet_contents
+
+/obj/structure/wall_closet/Initialize(mapload)
+	. = ..()
+	Initalize_closet_storage()
+	if(mapload)
+		return INITIALIZE_HINT_LATELOAD
+
+/obj/structure/wall_closet/LateInitialize()
+	Pickup_items()
+
+/obj/structure/wall_closet/proc/Initalize_closet_storage()
+	closet_contents = list()
+	for(var/i = 1, i <= 20, i++)
+		var/list/item_entry = list()
+		closet_contents += list(item_entry)
+
+/obj/structure/wall_closet/proc/Pickup_items()
+	var/atom/L = drop_location()
+	for(var/obj/item/I in L)
+		if(!Closet_insert_item(I, update_icons = FALSE))
+			break
+
+/obj/structure/wall_closet/proc/Closet_insert_item(obj/item/inserted_item, ui_index, update_icons = TRUE)
+	if(contents.len >= 20)
+		return FALSE
+	if(!ui_index)
+		for(var/index = 1, index <= closet_contents.len, index++)
```

Comment:
Use ranged for loops. This review applies to all for loops in this PR using the C style.
```
for(var/i in 1 to 20)
```

Reply:
adressed (i believe)